util/src/main/java/org/killbill/billing/util/entity/dao/EntitySqlDaoWrapperInvocationHandler.java

```diff
@@ -222,97 +210,6 @@ public Object execute() throws Throwable {
         });
     }
 
-    private Object invokeWithCaching(final Cachable cachableAnnotation, final Method method, final Object[] args)
```

Comment:
Here's my understanding of the current code (let me know if I'm missing something, it's a bit tricky...):

* `invokeWithCaching` is invoked only if `@Cachable` annotation is present
  * IIRC we used this annotation more in the past, but now we invoke the caches directly as the stuff we cache is more complex (e.g. see `InternalCallContextFactory`)
* `@Cachable` is annotated only on `AuditSqlDao` and `EntitySqlDao` methods
  1. For `AuditSqlDao`, `getObjectType()` would return `null` (as `AuditSqlDao` doesn't extend `EntitySqlDao`) 
  2. For `EntitySqlDao`, `getRecordId` was only used in `checkEntityAlreadyExists` (fixed by invoking the cache directly)

=> Hence, it seems largely unused.

Note that I believe we could maybe fix 1. above, by extracting the `tableName` (so the `ObjectType`) from the arguments of the `AuditSqlDao` methods (instead of looking at the class hierarchy), but I'm not sure we should bother. The multi-node scenario would still be broken and we somehow passed all of our load tests so far without this caching working...

Reply:
All good. Agreed it's a path that seems unused and less code is better...